include/af/device.h

```diff
@@ -75,10 +75,21 @@ namespace af
     ///
     /// \param[in] device the ID of the device to query
     ///
-    /// \returns true if the \p device supports double precision operations. false otherwise
+    /// \returns true if the \p device supports double precision operations.
+    ///          false otherwise
     /// \ingroup device_func_dbl
     AFAPI bool isDoubleAvailable(const int device);
 
+    /// \brief Queries the current device for half precision floating point
+    ///        support
+    ///
+    /// \param[in] device the ID of the device to query
+    ///
+    /// \returns true if the \p device supports half precision operations.
+    ///          false otherwise
+    /// \ingroup device_func_half
+    AFAPI bool isHalfAvailable(const int device);
```

Comment:
Could you add more comments in order to give some examples of which device will return true which false ?

Reply:
On my side I m more interested to know if the device can handle half natively with better perf than f32 (most of the time meaning Volta/Turing via TensorCores and apparently P100). 
Is nt that method going to return true on old GPUs (GTX10X0) where half is perhaps "supported" but perf is disastrous ?

Reply:
The half support for the 10x0 cards are supported through floats.

Even on the Volta cards we need to improve the way we handle things to improve the compute performance. In order to fully utilize the GPU's half compute performance we need to use half2. This will require a major shift on some of the internal kernels so I pushed that change off to later. We can talk about what functions are important to you and improve those later down the line. All the library calls for half(i.e. matmul) will be using half2 so you will get full compute performance there. If there are kernels that are bandwidth bound moving to half2 will serve no purpose because it will still be a bandwidth bound kernel.